### DISCUSSION TITLE
error: ‘deserialize_tag’ was not declared in this scope

### DISCUSSION THREAD

**Discussion by Jack012a** [simdjson/simdjson#2303] [General]
> simdjson::deserialize_tag: it is the tag for Customization Point Object (CPO). You may often ignore this parameter. It is used to indicate that you mean to provide a deserialization function for simdjson.

```
#include <iostream>
#include <string_view>
#include <vector>

#include "simdjson.h"

struct Car {
  std::string make;
  std::string model;
  int year;
  std::vector<float> tire_pressure;
};

namespace simdjson {
// This tag_invoke MUST be inside simdjson namespace
template <typename simdjson_value>
auto tag_invoke(deserialize_tag, simdjson_value &val, Car& car) {
  ondemand::object obj;
  auto error = val.get_object().get(obj);
  if (error) {
    return error;
  }
  if ((error = obj["make"].get_string(car.make))) {
    return error;
  }
  if ((error = obj["model"].get_string(car.model))) {
    return error;
  }
  if ((error = obj["year"].get(car.year))) {
    return error;
  }
  if ((error = obj["tire_pressure"].get<std::vector<float>>().get(
           car.tire_pressure))) {
    return error;
  }
  return simdjson::SUCCESS;
}
} // namespace simdjson

int main() {
  simdjson::padded_string json =
      R"( { "make": "Toyota", "model": "Camry",  "year": 2018,
       "tire_pressure": [ 40.1, 39.9 ] })"_padded;
  simdjson::ondemand::parser parser;
  simdjson::ondemand::document doc = parser.iterate(json);
  Car c(doc);
  std::cout << c.make << std::endl;
  return 0;
}

$ g++ -std=c++2b -o ./bin/type type.cpp simdjson.cpp -g -O0

type.cpp:17:17: error: ‘deserialize_tag’ was not declared in this scope
   17 | auto tag_invoke(deserialize_tag, simdjson_value &val, Car& car) {
```

**Question**> What should I change to fix this build issue?

Thank you

**Comment by lemire** [comment]
> What should I change to fix this build issue?

What is the `simdjson.cpp` version you are using?

The `deserialize_tag` is currently not part of a release.

For production use, I recommend working from a release. Each release comes with its documentation. E.g., you can grab the last release here:

https://github.com/simdjson/simdjson/archive/refs/tags/v3.10.1.zip

I am currently working on releasing 3.11 which will have this feature. I might release it tomorrow or soon after.

If you want to use it prior to a release (e.g., you want to work from our main branch, which is a development branch), you can do so, of course (although, please, not in production). Just make sure that whatever code you are calling as the matching features.

**Reply by Jack012a** [reply]
Hello Daniel,

I am using the latest release which is the version of 3.10.1. The code above is listed on the website as for simdjson [3.10.1](https://simdjson.github.io/simdjson/md_doc_basics.html).

Thank you for your update.

**Reply by lemire** [reply]
The website displays the current (main/development) version of the library. If you want the documentation of the release you are using, you need to look at the markdown files that accompany the source files.

**Reply by lemire** [reply]
We have a new release: https://github.com/simdjson/simdjson/releases/tag/v3.11.0

Consider updating.
